clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -407,7 +407,7 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                     log.debug("Ignoring fetched data for partition {} which is no longer assigned.", tp);
                 } else if (partition.errorCode == Errors.NONE.code()) {
                     long fetchOffset = request.fetchData().get(tp).offset;
-
+                    subscriptions.invokeConsumerSeekCallback(tp, null);
```

Comment:
Shouldn't this be invoked only after we've checked that this consume request matches the expected offset? I think the current code will behave incorrectly given the following sequence of events:

1. Fetch request sent
2. Rebalance
3. Seek
4. Fetch response triggers this code

The callback is invoked, but the fetch request was for the previous generation and errors like OFFSET_OUT_OF_RANGE will not be correctly passed to the callback.